csharp/src/Ice/IceDiscovery/Lookup.cs

```diff
@@ -34,13 +27,14 @@ internal class Lookup : ILookup
                 return; // Ignore
             }
 
-            (IObjectPrx? proxy, bool isReplicaGroup) = _registry.FindAdapter(adapterId, protocol ?? Protocol.Ice1);
+            (IObjectPrx? proxy, bool isReplicaGroup) = _registryServant.FindAdapter(adapterId);
             if (proxy != null)
             {
                 // Reply to the multicast request using the given proxy.
                 try
                 {
-                    reply.FoundAdapterByIdAsync(adapterId, proxy, isReplicaGroup, cancel: cancel);
+                    await reply.FoundAdapterByIdAsync(adapterId, proxy, isReplicaGroup, cancel: cancel).
+                        ConfigureAwait(false);
```

Comment:
Should we maybe catch and log a message here if we're unable to return a response?

Reply:
Fixed.